ambry-utils/src/main/java/com.github.ambry.utils/ZeroBytesInputStream.java

```diff
@@ -48,6 +48,12 @@ public int read() throws IOException {
 
   @Override
   public int read(byte[] bytes, int offset, int length) throws IOException {
+    if (length == 0) {
+      return 0;
+    }
```

Comment:
there are a few checks that the `InputStream#read()` is supposed to do as per the docs. Most implementations that override this method copy that verbatim:
```java
        if (b == null) {
            throw new NullPointerException();
        } else if (off < 0 || len < 0 || len > b.length - off) {
            throw new IndexOutOfBoundsException();
        } else if (len == 0) {
            return 0;
        }
```
I checked `ByteBufferInputStream#read()` and for that the bounds check happens anyway within `ByteBuffer#get()`, so this check is unnecessary. But for this class, it is necessary.